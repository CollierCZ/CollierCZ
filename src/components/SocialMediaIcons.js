import { css } from '@emotion/core'
import React from "react"
import { SocialIcon } from "react-social-icons"

class SocialMediaIcons extends React.Component {
  render() {
    const { urls, color } = this.props;
    if (urls && urls.length > 0) {
      return (
        <div
          css={css`
            text-align: left;
            margin: 0;
            .social-svg {
              text-align: center;
              color: rgb(255, 255, 255, 0.8);
            }
            .social-svg:hover {
              color: #bbc7cc;
              background: white;
            }
          `}
        >
          {urls.map(url => (
            <SocialIcon
              key={url}
              className="social-media-icon"
              url={url}
              color={color}
              style={{ height: null, width: null }}
              css={css`
                display: inline-block;
                margin: 0 1rem;
                width: 40px;
                height: 40px;
                z-index:2;  
                transition: all ease 0.3s;
                
                @media only screen and (max-width: 900px) {
                  display: inline-block;
                  width: 35px;
                  height: 35px;
                }
                @media only screen and (max-width: 500px) {
                  margin: 1rem;
                  display: inline-block;
                  width: 30px;
                  height: 30px;
                }
              `}
            />
          ))}
        </div>
      );
    }
    return null;
  }
}

export default SocialMediaIcons;
