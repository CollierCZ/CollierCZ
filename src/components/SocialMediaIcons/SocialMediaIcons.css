.social-media-icons {
    text-align: left;
    margin: 0;
}

.social-media-icons .social-svg {
    text-align: center;
}

.social-media-icon {
    display: inline-block;
    margin: 0 1rem;
    width: 40px;
    height: 40px;
    z-index:2;
}

/* Add subtle load-in animation for content on the home page */
.main-header .social-media-icon {
    -webkit-animation: fade-in-down 0.9s;
    animation: fade-in-down 0.9s both;
    -webkit-animation-delay: 0.1s;
    animation-delay: 0.1s;
}

.main-header .social-media-icon .social-svg {
    color: rgb(255, 255, 255, 0.8);
}

.main-header .social-media-icon .social-svg:hover {
    color: #bbc7cc;
    background: white;
}

/* ==========================================================================
       Media Queries - Smaller than 900px
   ========================================================================== */

@media only screen and (max-width: 900px) {
    .social-media-icon {
        display: inline-block;
        margin: 1.4rem 0.9rem 0;
        width: 40px;
        height: 40px;
    }
}

/* ==========================================================================
       Media Queries - Smaller than 500px
   ========================================================================== */

@media only screen and (max-width: 500px) {
    .social-media-icon {
        display: inline-block;
        margin: 1.2rem 0.8rem 0;
        width: 35px;
        height: 35px;
    }
}
