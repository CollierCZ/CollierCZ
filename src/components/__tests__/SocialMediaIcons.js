import React from "react"
import { shallow } from "enzyme"
import SocialMediaIcons from "../SocialMediaIcons"

const urlArray = ["mailto:aaron@collier.cz","https://github.com/CollierCZ","https://collier.cz/rss.xml"]

describe("SocialMediaIcons", () => {
  it("renders correctly with URLs", () => {
    const component = shallow (<SocialMediaIcons urls={urlArray} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly when empty", () => {
    const component = shallow (<SocialMediaIcons />)
    expect(component).toMatchSnapshot()
  })
})