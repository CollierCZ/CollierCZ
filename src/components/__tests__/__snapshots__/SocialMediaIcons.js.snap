// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaIcons renders correctly with URLs 1`] = `
.c1 {
  width: 32px;
  height: 32px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  fill: currentColor;
}

.c0 {
  display: inline-block;
  margin: 0 0.5rem;
  height: 44px;
  width: 44px;
  background-color: #FFFFFF;
  color: #252A31;
  z-index: 2;
  -webkit-transition: all ease 0.3s;
  transition: all ease 0.3s;
  border-radius: 50%;
}

.c0:hover {
  background-color: #252A31;
  color: #FFFFFF;
}

.c0 * {
  padding-top: 6px;
}

<div
  css={
    Array [
      "
    display:flex;
    ",
    ]
  }
>
  <a
    aria-label="mailto"
    className="c0 icon"
    href="mailto:aaron@collier.cz"
  >
    <svg
      className="c1"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.262 14.127l-8.07-8.072a.42.42 0 01-.119-.235.418.418 0 01.353-.473c.08-.011.16-.017.24-.017h16.667c.08 0 .161.006.24.017a.426.426 0 01.235.119.417.417 0 010 .589l-8.073 8.072a1.044 1.044 0 01-1.473 0zM21.74 6.889a.15.15 0 01.166-.032.15.15 0 01.093.14v10c0 .92-.746 1.667-1.667 1.667H3.667c-.92 0-1.667-.747-1.667-1.667v-10a.152.152 0 01.093-.138.15.15 0 01.167.032l5.167 5.167a.21.21 0 010 .295l-3.202 3.202a.624.624 0 000 .869c.24.249.634.255.883.015l3.202-3.202a.21.21 0 01.295 0l1.776 1.772a2.293 2.293 0 003.24 0l1.775-1.775c.08-.08.212-.08.294.001l3.203 3.203.015.016c.248.239.644.231.884-.017a.625.625 0 00-.015-.883l-3.203-3.203a.208.208 0 010-.295l5.167-5.167z"
      />
    </svg>
  </a>
  <a
    aria-label="github"
    className="c0 icon"
    href="https://github.com/CollierCZ"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-github fa-w-16 social-media-icon"
      css={
        Array [
          "
          width: 32px !important;
          height: 32px;
          ",
        ]
      }
      data-icon="github"
      data-prefix="fab"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <a
    aria-label="rss"
    className="c0 icon"
    href="https://collier.cz/rss.xml"
  >
    <svg
      className="c1"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 24 24"
    >
      <path
        d="M19.493 16.583a.833.833 0 010 1.667h-15a.833.833 0 110-1.667c.69 0 1.25-.56 1.25-1.25v-4.051A6.631 6.631 0 0111.16 4.56V2.833a.833.833 0 111.667 0v1.728a6.631 6.631 0 015.416 6.72v4.052c0 .69.56 1.25 1.25 1.25zM13.857 19.5a.207.207 0 01.206.185 2.083 2.083 0 11-4.153.232 1.934 1.934 0 01.013-.232.208.208 0 01.207-.185h3.727z"
      />
    </svg>
  </a>
</div>
`;
